### PR TITLE
Fix trailing commas on `Requires-Python` in HTML indexes 

### PR DESCRIPTION
**Pull request by davidszotten**
illustration and suggested fix for #1464

### CODE REVIEW

**Comment by charliermarsh** [comment]
We tend to fix this via our `lenient_requirements.rs` thing -- can you take a look at that file?

**Comment by davidszotten** [comment]
ah i see. how does this look?

**Comment by charliermarsh** [comment]
Ahh yes, this looks good.

**Review by charliermarsh — APPROVED** [review]
Thanks!

File: crates/uv-client/src/html.rs [hunk 1]
```diff
@@ -137,7 +138,7 @@ impl SimpleHtml {
         {
             let requires_python = std::str::from_utf8(requires_python.as_bytes())?;
             let requires_python = html_escape::decode_html_entities(requires_python);
-            Some(LenientVersionSpecifiers::from_str(&requires_python).map(Into::into))
+            Some(LenientVersionSpecifiers::from_str(&requires_python).map(VersionSpecifiers::from))
```
Comment (davidszotten) [inline crates/uv-client/src/html.rs:141]: how come from is preferable? (to help me learn)

**Pull request merged**